src/dialogs/config-flow/step-flow-abort.ts

```diff
@@ -33,6 +53,36 @@ class StepFlowAbort extends LitElement {
     `;
   }
 
+  private async _handleMissingCreds() {
+    if (
+      !(await showConfirmationDialog(this, {
+        title: this.hass.localize(
+          "ui.panel.config.integrations.config_flow.missing_credentials",
+          {
+            integration: domainToName(this.hass.localize, this.domain),
+          }
+        ),
+      }))
+    ) {
+      this._flowDone();
+      return;
+    }
+
+    showAddApplicationCredentialDialog(this, {
+      selectedDomain: this.domain,
+      dialogAbortedCallback: () => {
+        this._flowDone();
+      },
+      applicationCredentialAddedCallback: () => {
+        showConfigFlowDialog(this, {
```

Comment:
Oh interesting, you keep the element around so you can open dialogs on it. That feels a bit weird?

Reply:
Would you mind being a bit more perspective about the solution? I'm having trouble either pattern matching or learning lit element norms from trivial examples in docs. I'm not understanding how these events work and what they're event doing.

Does this need to happen in the parent element? somehow create a new element?

Reply:
To manage the stack of elements being drawn, we're opening dialogs by adding their elements at the root of the page, below the current app. We use events to talk to the [dialog-manager-mixin](https://github.com/home-assistant/frontend/blob/dev/src/state/dialog-manager-mixin.ts).

When inside the HA app, this can alternatively be retrieved with `document.querySelector('home-assistant')`. However, this dialog can also be shown during onboarding 👀 (not sure if that would even work, easy to test by pointing HA at a new config dir (`mkdir config2 && hass -c config2`))

Reply:
The parent element of thet step is also going to be removed, you need to go to the parent of the dialog element (`<dialog-data-entry-flow>`).

Reply:
This now explicitly passes up the parent dialog in order to use it as the input element to the dialog and restarting the config flow.